go/cli/mcap/cmd/cat.go

```diff
@@ -109,57 +109,68 @@ func printMessages(
 			}
 			continue
 		}
-		switch schema.Encoding {
-		case "ros1msg":
-			transcoder, ok := transcoders[channel.SchemaID]
-			if !ok {
-				packageName := strings.Split(schema.Name, "/")[0]
-				transcoder, err = ros.NewJSONTranscoder(packageName, schema.Data)
-				if err != nil {
-					return fmt.Errorf("failed to build transcoder for %s: %w", channel.Topic, err)
+		if schema == nil || schema.Encoding == "" {
```

Comment:
Schemaless channels should not have a schema record associated.
```suggestion
		if schema == nil {
```

Reply:
Well, actually, at least according to https://github.com/foxglove/mcap/blob/0842cbf26dbf114a0fa0eb2dc36355f6db75382f/python/mcap/mcap/well_known.py#L20

This schemaless == `SelfDescribing` and can have `""` as a valid tag.

Reply:
My mistake, you're right.